src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -188,6 +188,13 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     mWiFiState          = WIFI_STATE_SCANNING;
     mSsidFound          = false;
 
+    if ((ssid.size() > 0) && (!mInternalScan))
+    {
+        mDirectScanNetwork.Erase();
```

Comment:
I guess we could reuse `mWantedNetwork` and simplify the implementation but doesn't matter for now ;)